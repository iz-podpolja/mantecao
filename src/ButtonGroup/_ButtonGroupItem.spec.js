import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ButtonGroupItem from './_ButtonGroupItem'

describe('ButtonGroupItem Component', () => {
  it('renders a first button properly', () => {
    const tree = renderer
      .create(
        <ButtonGroupItem
          first={true}
          last={false}
          onSelect={() => false}
          selected={false}
          text={'whatup'}
          value={1}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a middle button properly', () => {
    const tree = renderer
      .create(
        <ButtonGroupItem
          first={false}
          last={false}
          onSelect={() => false}
          selected={false}
          text={'whatup'}
          value={1}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a last button properly', () => {
    const tree = renderer
      .create(
        <ButtonGroupItem
          first={false}
          last={true}
          onSelect={() => false}
          selected={false}
          text={'whatup'}
          value={1}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders properly disabled', () => {
    const tree = renderer
      .create(
        <ButtonGroupItem
          disabled
          first={false}
          last={true}
          onSelect={() => false}
          selected={false}
          text={'whatup'}
          value={1}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('fires onChange event properly', () => {
    const spy = jest.fn()
    const fakeValue = 1
    const component = shallow(
      <ButtonGroupItem
        disabled
        first={false}
        last={true}
        onSelect={spy}
        selected={false}
        text={'whatup'}
        value={fakeValue}
      />
    )
    component.simulate('click')
    expect(spy.mock.calls[0][0]).toEqual(fakeValue)
  })
})
