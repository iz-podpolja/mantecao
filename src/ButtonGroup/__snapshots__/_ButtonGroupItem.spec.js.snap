// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroupItem Component renders a first button properly 1`] = `
<button
  className="sc-bdVaJa dNukzi"
  color="#CCCCCC"
  disabled={undefined}
  onClick={[Function]}
  style={
    Object {
      "borderBottomLeftRadius": "4px",
      "borderBottomRightRadius": "0px",
      "borderLeft": "1px solid #b3b3b3",
      "borderTopLeftRadius": "4px",
      "borderTopRightRadius": "0px",
      "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
      "padding": ".5em 1.5em",
    }
  }
>
  whatup
</button>
`;

exports[`ButtonGroupItem Component renders a last button properly 1`] = `
<button
  className="sc-bdVaJa dNukzi"
  color="#CCCCCC"
  disabled={undefined}
  onClick={[Function]}
  style={
    Object {
      "borderBottomLeftRadius": "0px",
      "borderBottomRightRadius": "4px",
      "borderLeft": "1px solid #b3b3b3",
      "borderTopLeftRadius": "0px",
      "borderTopRightRadius": "4px",
      "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
      "padding": ".5em 1.5em",
    }
  }
>
  whatup
</button>
`;

exports[`ButtonGroupItem Component renders a middle button properly 1`] = `
<button
  className="sc-bdVaJa dNukzi"
  color="#CCCCCC"
  disabled={undefined}
  onClick={[Function]}
  style={
    Object {
      "borderBottomLeftRadius": "0px",
      "borderBottomRightRadius": "0px",
      "borderLeft": "1px solid #b3b3b3",
      "borderTopLeftRadius": "0px",
      "borderTopRightRadius": "0px",
      "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
      "padding": ".5em 1.5em",
    }
  }
>
  whatup
</button>
`;

exports[`ButtonGroupItem Component renders properly disabled 1`] = `
<button
  className="sc-bdVaJa dNukzi"
  color="#CCCCCC"
  disabled={true}
  onClick={[Function]}
  style={
    Object {
      "borderBottomLeftRadius": "0px",
      "borderBottomRightRadius": "4px",
      "borderLeft": "1px solid #b3b3b3",
      "borderTopLeftRadius": "0px",
      "borderTopRightRadius": "4px",
      "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
      "padding": ".5em 1.5em",
    }
  }
>
  whatup
</button>
`;
